From 796464015bffe6e0beca9e1c355b223512803c3e Mon Sep 17 00:00:00 2001
From: Mingli Yu <mingli.yu@windriver.com>
Date: Wed, 27 Feb 2019 23:01:00 -0800
Subject: [PATCH] sql/CMakeLists.txt: fix gen_lex_hash not found

Fix the below do_compile issue in cross-compiling env.
| make[2]: *** No rule to make target '/build/tmp/work/aarch64-poky-linux/mariadb/10.3.13-r0/mariadb-10.3.13/sql/gen_lex_hash', needed by 'sql/lex_hash.h'.  Stop.
| make[2]: *** No rule to make target '/build/tmp/work/aarch64-poky-linux/mariadb/10.3.13-r0/mariadb-10.3.13/sql/gen_lex_token', needed by 'sql/lex_token.h'.  Stop.

Upstream-Status: Inappropriate [oe build specific]

Signed-off-by: Mingli Yu <mingli.yu@windriver.com>

---
 sql/CMakeLists.txt | 30 ++++++++++++++++++++----------
 1 file changed, 20 insertions(+), 10 deletions(-)

Index: mariadb-10.4.17/sql/CMakeLists.txt
===================================================================
--- mariadb-10.4.17.orig/sql/CMakeLists.txt
+++ mariadb-10.4.17/sql/CMakeLists.txt
@@ -55,11 +55,16 @@ ${CMAKE_BINARY_DIR}/sql
 
 
 
-ADD_CUSTOM_COMMAND(
-  OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/lex_token.h
-  COMMAND gen_lex_token > lex_token.h
-  DEPENDS gen_lex_token
-)
+IF(NOT CMAKE_CROSSCOMPILING)
+  ADD_CUSTOM_COMMAND(
+   OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/lex_token.h
+   COMMAND gen_lex_token > lex_token.h
+   DEPENDS gen_lex_token)
+ELSE()
+  ADD_CUSTOM_COMMAND(
+   OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/lex_token.h
+   COMMAND gen_lex_token > lex_token.h)
+ENDIF()
 
 ADD_DEFINITIONS(-DMYSQL_SERVER -DHAVE_EVENT_SCHEDULER)
 
@@ -364,11 +369,16 @@ IF(NOT CMAKE_CROSSCOMPILING)
   ADD_EXECUTABLE(gen_lex_hash gen_lex_hash.cc)
 ENDIF()
 
-ADD_CUSTOM_COMMAND(
-  OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/lex_hash.h
-  COMMAND gen_lex_hash > lex_hash.h
-  DEPENDS gen_lex_hash
-)
+IF(NOT CMAKE_CROSSCOMPILING)
+  ADD_CUSTOM_COMMAND(
+   OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/lex_hash.h
+   COMMAND gen_lex_hash > lex_hash.h
+   DEPENDS gen_lex_hash)
+ELSE()
+  ADD_CUSTOM_COMMAND(
+   OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/lex_hash.h
+   COMMAND gen_lex_hash > lex_hash.h)
+ENDIF()
 
 MYSQL_ADD_EXECUTABLE(mysql_tzinfo_to_sql tztime.cc COMPONENT Server)
 SET_TARGET_PROPERTIES(mysql_tzinfo_to_sql PROPERTIES COMPILE_FLAGS "-DTZINFO2SQL")
